fastembed/common/model_management.py

```
@@ -304,20 +304,44 @@ def decompress_to_cache(cls, targz_path: str, cache_dir: str) -> str:
+                try:
+                    tar.extractall(path=cache_dir, members=members, filter="data")
+                except TypeError:  # pragma: no cover - Python < 3.12
+                    tar.extractall(path=cache_dir, members=members)
+        except (tarfile.TarError, ValueError) as e:
             # If any error occurs while opening or extracting the tar.gz file,
             # delete the cache directory (if it was created in this function)
             # and raise the error again
-            if "tmp" in cache_dir:
+            if "tmp" in cache_dir and os.path.exists(cache_dir):
                 shutil.rmtree(cache_dir)
             raise ValueError(f"An error occurred while decompressing {targz_path}: {e}")
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)